src/app/shared/components/product/components/product-item/product-item.component.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ProductItemComponent } from './product-item.component';
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-test-host',
+  template: `
+    <app-product-item>
+      <p id="testId" slot="id">Test ID</p>
+      <img id="testImg" src="test.jpg" alt="test" />
+      <h3 id="testTitle">Test Title</h3>
+      <p id="testDescription" slot="description">Test Description</p>
+      <div id="testDiv">Should not be projected</div>
+    </app-product-item>
+  `,
+})
+class TestHostComponent {}
+
+describe('ProductItemComponent', () => {
+  let component: TestHostComponent;
+  let fixture: ComponentFixture<TestHostComponent>;
+  let compiled: HTMLElement;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ProductItemComponent, ProductItemComponent],
+      declarations: [TestHostComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    compiled = fixture.debugElement.nativeElement;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should project <p>, <img>, <h3> elements and ignore other elements', () => {
+    const idElement = compiled.querySelectorAll('#testId');
```

Comment:
You should always validate against the selectors that you defined in the component and not against your custom id's.